dandischema/models.py

```diff
@@ -200,6 +200,8 @@ def schema_extra(schema: Dict[str, Any], model) -> None:
                     value["title"] = name2title(prop)
                 if value.get("format", None) == "uri":
                     value["maxLength"] = 1000
+                if value.get("pattern", "").startswith("^http"):
```

Comment:
`"pattern"` here a string, not regex,  then `^` wouldn't work. also some smartypants could have "httplike". Lets do

```suggestion
                if re.match("https?://", value.get("pattern", "")):
```
but needs `import re` on top (no there yet).

Reply:
pattern is a regex, so the match will work. if you look at all the regex' which translate into "pattern" for schema generation they all have the start of string character. but i like this change. i'll commit this and then add re.